### PR TITLE
Enable specifying ENS names as contact codes

### PR DESCRIPTION
**Pull request by jrainville**
I had to add redux-thunk, because getting the ENS address in the action is way easier. Let me know if I shouldn't or there is a better way, but I couldn't find a good way to to it in the saga (especially since we use `doTransaction`).

### CODE REVIEW

**Comment by jrainville** [comment]
@alaibe I don't mind using saga, but I really don't know an easy way to do that. Like I said in the description, since we sue `doTransaction`, it's really hard to actually modify the payload before sending.

Maybe we can pair to find a way. Otherwise, I feel like the `thunk` solution really is not that bad.

File: app/js/utils/saga.js [hunk 1]
```diff
@@ -44,3 +45,22 @@ export function *doTransaction(preSuccess, success, failed, {value = 0, toSend})
     yield put({type: failed, error: error.message});
   }
 }
+
+export function getEnsAddress(name) {
```
Comment (alaibe) [inline app/js/utils/saga.js]: there is an embark js service, that would probably be a better location

Comment (alaibe) [reply app/js/utils/saga.js]: We probably need to update the contract to allow non address?

**Comment by alaibe** [comment]
@jrainville see my latest commit on this PR and tell me if it works.
I have nothing against  `thunk`  and I am fine moving to it if you feel it is simpler. 
Having the 2 together would be like having `underscore` and `lodash` => bigger build...

**Comment by jrainville** [comment]
You are the best @alaibe, your solution is very good!

https://i.redd.it/fskhv6es09v01.png

**Comment by jrainville** [comment]
Updated without Thunk!

File: app/js/utils/saga.js (hunk 1 above)
Comment (jrainville) [reply app/js/utils/saga.js]: Moved. Also, update the contract to be a string instead of an address?

**Pull request merged**